File: src/hashmap/lc_49_group_anagrams/group_anagrams.py
from collections import defaultdict
class Solution:
    def groupAnagrams(self, strs: list[str]) -> list[list[str]]:
        res = []
        hashmap = defaultdict(list)
        for st in strs:
            stsorted = ''.join(sorted(st))
            hashmap[stsorted].append(st)
        for key in hashmap:
            res.append(hashmap[key])
        return res

File: src/hashmap/lc_49_group_anagrams/group_anagrams.java
import java.util.*;
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        List<List<String>> res = new ArrayList<>();
        Map<String, List<String>> map = new HashMap<>();
        for(String st : strs) {
            char[] charArray = st.toCharArray();
            Arrays.sort(charArray);
            String sortedString = new String(charArray);
            map.computeIfAbsent(sortedString, key -> new ArrayList<>()).add(st);
        }
        for(Map.Entry<String, List<String>> entry : map.entrySet()) {
            res.add(entry.getValue());
        }
        return res;
    }
}